lucille-core/src/main/java/com/kmwllc/lucille/core/Runner.java

```diff
@@ -162,43 +162,54 @@ public static Map<String, TestMessenger> runInTestMode(String configName) throws
     return result.getHistory();
   }
 
+  // Returns a mapping from pipeline names to the list of exceptions produced when validating them.
   private static Map<String, List<Exception>> logValidation(Config config) throws Exception {
```

Comment:
Suggest that this method should be typed as `public static void logValidation(Map<String, List<Exception>> exceptionMap)` and be dedicated only to logging (and stringifying in order to do logging). Some changes would be needed in a few places where logValidation is called, to get the validation result and pass it to this method.